sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/BlobUserAgentModificationPolicy.java

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.implementation.util;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpPipelinePosition;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Mono;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This policy modifies the blob user agent string for clients created in packages that are dependencies of blob.
+ * It transforms a User Agent String as follows
+ * UAbefore: "azsdk-java-azure-storage-blob/12.11.0-beta.2 (11.0.6; Windows 10; 10.0)"
+ * UAafter: "azsdk-java-azure-storage-blob/12.11.0-beta.2 azsdk-java-azure-storage-blob-batch/12.8.0-beta.2 (11.0.6; Windows 10; 10.0) "
+ */
+public class BlobUserAgentModificationPolicy implements HttpPipelinePolicy {
+
+    private final String clientName;
+    private final String clientVersion;
+
+    private static final String USER_AGENT = "User-Agent";
+    private static final String REGEX = "azsdk-java-azure-storage-blob/\\d+\\.\\d+\\.\\d+[-beta\\.\\d+]*(.)*";
+    private static final Pattern PATTERN = Pattern.compile(REGEX);
+
+    /**
+     * Creates a new BlobUserAgentModificationPolicy.
+     *
+     * @param clientName The name of the package.
+     * @param clientVersion The version of the package.
+     */
+    public BlobUserAgentModificationPolicy(String clientName, String clientVersion) {
+        this.clientName = clientName;
+        this.clientVersion = clientVersion;
+    }
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+
+        String userAgent = context.getHttpRequest().getHeaders().getValue(USER_AGENT);
+        Matcher matcher = PATTERN.matcher(userAgent);
+        StringBuilder builder = new StringBuilder();
+        if (matcher.matches()) {
```

Comment:
Do we only want to apply the modification if the regex matches the entire User-Agent or only if it contains it? As the current regex doesn't support prepended custom application IDs: https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/util/UserAgentUtil.java#L40

Reply:
Good point. Can you take a look again and see if this accounts for that? I loosened the regex and insert after the location of the found regex